### PR TITLE
Log more info about internal errors

### PR DESCRIPTION
**Pull request by alexmojaki**
Replaces https://github.com/pydantic/logfire/pull/325

### CODE REVIEW

File: logfire/_internal/utils.py [hunk 1]
```diff
@@ -252,17 +256,83 @@ def log_internal_error():
...
+
+def _handle_internal_errors():
     try:
         yield
     except Exception:
         log_internal_error()
 
 
+_HANDLE_INTERNAL_ERRORS_CODE = _handle_internal_errors.__code__
+handle_internal_errors = contextmanager(_handle_internal_errors)
```
Comment (Kludex) [inline logfire/_internal/utils.py]: If you do this, I think the stub generator will work properly, and not generate the `Incomplete` - not sure tho. Can we try?
```suggestion
handle_internal_errors: ContextManager[None, None, None] = contextmanager(_handle_internal_errors)
```

File: logfire-api/logfire_api/_internal/utils.pyi [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from _typeshed import Incomplete
 from collections.abc import Generator
+from logfire._internal.stack_info import is_user_code as is_user_code
```
Comment (Kludex) [inline logfire-api/logfire_api/_internal/utils.pyi:3]: Why is the `as` needed?

File: logfire/_internal/utils.py [hunk 3]
```diff
@@ -252,7 +256,70 @@ def log_internal_error():
         raise
 
     with suppress_instrumentation():  # prevent infinite recursion from the logging integration
-        logger.exception('Internal error in Logfire')
+        logger.exception('Internal error in Logfire', exc_info=_internal_error_exc_info())
+
+
+def _internal_error_exc_info() -> Any:
```
Comment (Kludex) [inline logfire/_internal/utils.py]:
```suggestion
ExcInfoType = tuple[Type[BaseException] | None, BaseException | None, types.TracebackType | None]

def _internal_error_exc_info() -> ExcInfoType:
```

**Review by Kludex — APPROVED** [review]

File: logfire-api/logfire_api/_internal/utils.pyi (hunk 2 above)
Comment (alexmojaki) [reply logfire-api/logfire_api/_internal/utils.pyi:3]: That's just what stubgen does, you can see it in lots of other places in the pyi files

**Pull request merged**